dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.cpp

```diff
@@ -2626,38 +2633,71 @@ RtpsUdpDataLink::bundle_mapped_meta_submessages(const Encoding& encoding,
   }
 }
 
+void
+RtpsUdpDataLink::harvest_send_queue(const MonotonicTimePoint& /*now*/)
+{
+  sq_.ready_to_send();
+  sq_.begin_transaction();
+
+  disable_response_queue(true);
+}
+
 void
 RtpsUdpDataLink::flush_send_queue(const MonotonicTimePoint& /*now*/)
 {
   ACE_Guard<ACE_Thread_Mutex> fsq_guard(fsq_mutex_);
-  sq_.condense_and_swap(fsq_vec_);
-  bundle_and_send_submessages(fsq_vec_);
-  fsq_vec_.clear();
+  flush_send_queue_i();
+}
+
+void
+RtpsUdpDataLink::flush_send_queue_i()
+{
+  for (size_t idx = 0; idx != fsq_vec_size_; ++idx) {
+    dedup(fsq_vec_[idx]);
```

Comment:
If you're calling `sq_.end_transaction()`  every time `disable_response_queue()` is called, I think you wind up potentially getting a vector of submessages in `fsq_vec_` for every inbound packet. And if that's the case, calling `dedup()` and `bundle_and_send_submessages()` on them individually doesn't seem like it would let de-duplication or bundling help out all that much, since there wouldn't be much accumulated by that point.

Reply:
`end_transaction` will only produce a vector if `ready_to_send` has been called.  This occurs at the frequency of the harvest task so we should not get a vector per processed packet.